src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/util/RandomUtil.java

```diff
@@ -200,53 +212,242 @@ public static EmailHistory createEmailHistory() {
+		dispute.setDriversLicenceNumber(randomAlphanumeric(30));
+		dispute.setDriversLicenceProvince(randomAlphabetic(30));
+		dispute.setEmailAddress(randomAlphanumeric(100));
+		dispute.setFilingDate(randomDate());
+		dispute.setFineReductionReason(randomAlphabetic(500));
+		dispute.setHomePhoneNumber(randomNumeric(20));
+		dispute.setInterpreterLanguage(randomAlphabetic(10)); // TODO: what is the max length of this field?
+		dispute.setInterpreterRequired(randomYN());
+		dispute.setIssuedTs(randomDateTime());
+		dispute.setLawyerAddress(randomAlphanumeric(300));
+		dispute.setLawyerEmail(randomAlphanumeric(20)); // TODO: what is the max length of this field?
+		dispute.setLawyerGivenName1(randomAlphabetic(30));
+		dispute.setLawyerGivenName2(randomAlphabetic(30));
+		dispute.setLawyerGivenName3(randomAlphabetic(30));
+		dispute.setLawyerPhoneNumber(randomNumeric(20));
+		dispute.setLawyerSurname(randomAlphabetic(30));
+		dispute.setNoticeOfDisputeGuid(UUID.randomUUID().toString());
+		dispute.setOcrViolationTicket(randomAlphabetic(4000));
+		dispute.setOfficerPin(randomAlphabetic(10));
+		dispute.setPostalCode(randomAlphanumeric(10));
+		dispute.setRejectedReason(randomAlphabetic(500));
+		dispute.setRepresentedByLawyer(randomYN());
+		dispute.setStatus(DisputeStatus.NEW);
+		dispute.setSubmittedTs(randomDateTime());
+		dispute.setSystemDetectedOcrIssues(randomYN());
+		dispute.setTicketNumber(randomTicketNumber());
+		dispute.setTimeToPayReason(randomAlphabetic(500));
+		dispute.setUserAssignedTo(randomAlphabetic(30));
+		dispute.setUserAssignedTs(randomDateTime());
+		dispute.setViolationTicket(createViolationTicket(dispute));
+		dispute.setWitnessNo(Integer.valueOf(randomNumeric(3)));
+		dispute.setWorkPhoneNumber(randomNumeric(20));
+
+		// 36 character field
+		assertTrue(dispute.getNoticeOfDisputeGuid().length() == 36);
+
+		return dispute;
+	}
+
+	public static ViolationTicket createViolationTicket(Dispute dispute) {
+		ViolationTicket violationTicket = new ViolationTicket();
+		violationTicket.setAddress(randomAlphanumeric(30)); // TODO: what is the max length of this field?
+		violationTicket.setAddressCity(randomAlphanumeric(100));
+		violationTicket.setAddressCountry(randomAlphanumeric(100));
+		violationTicket.setAddressPostalCode(randomAlphanumeric(10));
+		violationTicket.setAddressProvince(randomAlphanumeric(100));
+		violationTicket.setCourtLocation(randomAlphanumeric(150));
+		violationTicket.setDetachmentLocation(randomAlphanumeric(150));
+		violationTicket.setDisputantOrganizationName(randomAlphabetic(150));
+		violationTicket.setDisputantSurname(randomAlphabetic(100));
+		violationTicket.setDisputantGivenNames(randomAlphabetic(200));
+		violationTicket.setDisputantDriversLicenceNumber(randomAlphanumeric(30));
+		violationTicket.setDisputantClientNumber(randomAlphanumeric(30));
+		violationTicket.setDriversLicenceProvince(randomAlphabetic(2)); // TODO: what is the max length of this field?

```

Comment:
This should be 100 characters as well based on field in the OCCAM table.

Reply:
Great, thanks. Updated.